framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -335,6 +343,65 @@ MeshDiagnosticsGenerator::checkWaterTightSidesets(const std::unique_ptr<MeshBase
   diagnosticsLog(message, _check_watertight_sidesets, num_faces_without_sideset);
 }
 
+void
+MeshDiagnosticsGenerator::checkWatertightNodesets(const std::unique_ptr<MeshBase> & mesh) const
+{
+  /*
+  Diagnostic Overview:
+  1) Mesh precheck
+  2) Loop through all elements
+  3) Loop through all sides of that element
+  4) If side is external loop through its nodes
+  5) If node is not associated with any nodeset add to list
+  6) Print out node id
+  */
+  if (mesh->mesh_dimension() < 2)
+    mooseError("The nodeset check only works for 2D and 3D meshes");
+  auto & boundary_info = mesh->get_boundary_info();
+  boundary_info.build_side_list();
+  const auto nodeset_map = boundary_info.get_nodeset_map();
```

Comment:
where is the _boundary_node_id being initialized in libmesh? is the previous call doing it?

Reply:
So I looked into it. The node ids get added to _boundary_node_id after execute_mesh_generators is called in ActionWarehouse.C. From what I can tell the mesh is always initialized before this test could ever be called.

Reply:
>  after execute_mesh_generators

wouldnt that be too late then?

Reply:
Sorry, it's the execute_mesh_generators task that initializes _boundary_node_id. After initialization mesh diagnostics runs. So that task runs both I believe, but it should initialize first.